core/trino-main/src/main/java/io/trino/simd/BlockEncodingSimdSupport.java

```diff
@@ -159,26 +164,30 @@ enum ArmSimdInstructionSet {
             }
         }
 
-        if (PREFERRED_BIT_WIDTH < MINIMUM_SIMD_LENGTH || !armFlags.contains(ArmSimdInstructionSet.sve)) {
+        // NEON support is often too slow to make this worthwhile
+        if (!armFlags.contains(ArmSimdInstructionSet.sve)) {
             return SimdSupport.NONE;
         }
 
-        // SVE 1 is sufficient to have Vector#compress(VectorMask) intrinsics for all primitive types that outperforms scalar code
-        boolean compressAll = armFlags.contains(ArmSimdInstructionSet.sve);
-        // SVE 2 has intrinsics for Vector#expand(VectorMask) over int and long, but not byte or short
-        boolean expandIntAndLong = armFlags.contains(ArmSimdInstructionSet.sve2);
-        // AARCH64 has intrinsics added to NEON, SVE1 and SVE2 in JDK 26 by https://bugs.openjdk.org/browse/JDK-8363989,
-        // reconsider enabling vectorized expand after testing with those changes
-        boolean expandByteAndShort = false;
+        // SVE 1 is sufficient to have Vector#compress(VectorMask) intrinsics for all primitive types
+        boolean compressAll = true;
+        boolean compressLong = preferredVectorBitWidth > 128; // only vectorize long compression if we can handle more than 2 values per instruction
+
+        // As of JDK 25, SVE 2 intrinsics for Vector#expand(VectorMask) over int and long, but not byte or short. JDK 26 add intrinsics
+        // for byte and short on SVE 2, SVE 1, and NEON in https://bugs.openjdk.org/browse/JDK-8363989. We can reconsider enabling vectorized
+        // expansion for those types at some point in the future
+        boolean expandInt = armFlags.contains(ArmSimdInstructionSet.sve2);
+        boolean expandLong = armFlags.contains(ArmSimdInstructionSet.sve2) && preferredVectorBitWidth > 128; // ensure minimum register width for long
+        boolean expandByteAndShort = false; // no intrinsics in JDK 25
```

Comment:
Can we print a log line which will summarize what SIMD capabilities were detected by the code ?
This has gotten complex enough that I suspect we will need that info for debugging some day.